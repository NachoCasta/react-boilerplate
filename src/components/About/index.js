export { default } from "./AboutView";
export * from "./AboutReducer";
