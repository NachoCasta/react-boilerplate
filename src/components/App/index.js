export { default } from "./App";
export { default as AppNavBar } from "./NavBar";
export { default as AppContainer } from "./Container";
