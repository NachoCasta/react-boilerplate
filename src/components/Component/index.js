export { default } from "./ComponentView";
export * from "./ComponentReducer";
