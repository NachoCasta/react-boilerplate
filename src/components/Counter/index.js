export { default } from "./CounterContainer";
export * from "./CounterReducer";
