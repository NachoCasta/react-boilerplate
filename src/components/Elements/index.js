export { default } from "./ElementsContainer";
export * from "./ElementsReducer";
