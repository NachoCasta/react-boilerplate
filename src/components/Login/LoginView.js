import React from "react";
import {
	Flex,
	Box,
	CardContainer,
	Heading,
	Button,
	Form,
	InputField
} from "components/UI";

const Login = ({ fields, onSubmit }) => {
	return (
		<CardContainer width={[1, 3 / 4, 2 / 3, 1 / 2]}>
			<Flex justifyContent="center">
				<Box>
					<Heading align="center">Login</Heading>
				</Box>
			</Flex>
			<Form>
				<Flex justifyContent="space-around">
					{fields.map(({ width, label, ...rest }) => (
						<Box key={label} width={width} p={1}>
							<InputField label={label} {...rest} />
						</Box>
					))}
					<Box width={1} p={1}>
						<Button type="button" width={1} onClick={onSubmit}>
							Login
						</Button>
					</Box>
				</Flex>
			</Form>
		</CardContainer>
	);
};

export default Login;
