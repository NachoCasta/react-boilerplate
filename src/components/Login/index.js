export { default } from "./LoginContainer";
export * from "./LoginReducer";
