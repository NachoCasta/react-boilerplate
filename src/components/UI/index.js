export * from "react-spinners";
export * from "./Springs";
export * from "./LinkList";
export * from "antd";
export {
    Container,
    Heading,
    Text,
    Link,
    RatingBadge,
    Hide
} from "pcln-design-system";
export { default as Box } from "./Box";
export { default as Flex } from "./Flex";
export { default as Card } from "./Card";
export { default as FadeCard } from "./FadeCard";
export { default as CardContainer } from "./CardContainer";
export { default as Form } from "./Form";
export { default as InputField } from "./InputField";
export { default as OnClickLink } from "./OnClickLink";
export { default as NavBar } from "./NavBar";
export { default as GradientBackgroundImage } from "./GradientBackgroundImage";
export { default as Input } from "./Input";
export { default as Select } from "./Select";
export { default as Button } from "./Button";
export { default as LoadingButton } from "./LoadingButton";
export { default as CircleButton } from "./CircleButton";
export { default as AddButton } from "./AddButton";
export { default as Scrollbar } from "./Scrollbar";
export { default as Code } from "./Code";
export { default as TextBlock } from "./TextBlock";
export { default as EditableTextBlock } from "./EditableTextBlock";
