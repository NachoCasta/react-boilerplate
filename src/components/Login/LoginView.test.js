import React from "react";
import { shallow, mount } from "enzyme";

import { Text, Button, OnClickLink } from "components/UI";
import LoginView from "./LoginView";

describe("LoginView", () => {
	it("SignIn matches snapshot", () => {
		const component = shallow(<LoginView fields={[]} signIn={true} />);
		expect(component).toMatchSnapshot();
	});

	it("SignUp matches snapshot", () => {
		const component = shallow(<LoginView fields={[]} signIn={false} />);
		expect(component).toMatchSnapshot();
	});

	it("has SignIn text when signIn set to true", () => {
		const component = mount(<LoginView fields={[]} signIn={true} />);
		expect(component.find(Text).text()).toContain(
			"Don't have an account yet?"
		);
		expect(component.find(Button).text()).toContain("Sign In");
		expect(component.find(OnClickLink).text()).toContain("Sign Up");
	});

	it("has SignUp text when signIn set to false", () => {
		const component = mount(<LoginView fields={[]} signIn={false} />);
		expect(component.find(Text).text()).toContain(
			"Already have an account?"
		);
		expect(component.find(Button).text()).toContain("Sign Up");
		expect(component.find(OnClickLink).text()).toContain("Sign In");
	});
});
