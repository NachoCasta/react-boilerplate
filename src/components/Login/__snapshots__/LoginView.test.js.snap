// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginView SignIn matches snapshot 1`] = `ShallowWrapper {}`;

exports[`LoginView SignUp matches snapshot 1`] = `ShallowWrapper {}`;
